workers/main/src/index.test.ts

```diff
@@ -26,13 +25,13 @@ describe('run', () => {
 
 describe('handleRunError', () => {
   it('should log the error and throw the error', () => {
-    const logSpy = vi
```

Comment:
You don't test a case when error is not instance if Error. I think it worth adding that.